### PR TITLE
fix: max tokens limit

### PR DESCRIPTION
**Pull request by joanamcsa**
**What type of PR is this?**

Fix: A bug fix

**What this PR does / why we need it:**

- Counts the number of tokens in a conversation and summarizes if it's bigger than 3000.

**Which issue(s) this PR fixes:**

Fixes [#52](https://github.com/xgeekshq/intelli-mate/issues/52)

**Special notes for your reviewer:**

After exceeding the limit number of tokens

image

### CODE REVIEW

File: packages/api/src/chats/usecases/add-message-to-chat.usecase.ts [hunk 1]
```diff
@@ -8,6 +8,8 @@ import { Usecase } from '@/common/types/usecase';
 import { ChatMessageResponseSchema } from '@/contract/chats/chat-message.response.dto';
 import { Injectable } from '@nestjs/common';
 
+const TOKENS_LIMIT = 3000;
```
Comment (comoser) [inline packages/api/src/chats/usecases/add-message-to-chat.usecase.ts:11]: this is going to live in our application configurations service. It reads from the `api/config/default.json`. It's going to be another one of the `ai` configurations

Comment (romansharapov19) [inline packages/api/src/chats/usecases/add-message-to-chat.usecase.ts:11]: Please add the `TOKEN_LIMIT` to the config file.

**Review by romansharapov19 — APPROVED** [review]
Nice job! 🐞 🔧

File: packages/api/src/ai/services/ai.service.ts [hunk 2]
```diff
@@ -16,13 +16,42 @@ export class AiService {
...
+    );
 
     try {
       return await chain.call({ input });
     } catch (e) {
       console.error(e.response.data);
     }
   }
+
+  async askAiToSummarize(chatId: string) {
```
Comment (comoser) [inline packages/api/src/ai/services/ai.service.ts:46]:
```suggestion
  private async askAiToSummarize(chatId: string) {
```

I don't want this method to be exposed to the outside world of the AI service. The AI service is the Facade API to whoever wishes to use it. For now the API exposes the `askAiInFreeText` method, and no other :)

**Review by comoser — APPROVED** [review]
amazing job 🚀 

left a few changes though

File: packages/api/src/chats/usecases/add-message-to-chat.usecase.ts (hunk 1 above)
Comment (joanamcsa) [reply packages/api/src/chats/usecases/add-message-to-chat.usecase.ts]: Yup, done here 22bbeb5acd58edcc08310a9f9035d455423b046a

**Pull request merged**